src/test/java/com/amplitude/api/AmplitudeClientTest.java

```diff
@@ -2493,4 +2493,63 @@ public void testMultipleIdentifyWithSetActionAndSetGroup() throws JSONException
         assertEquals(0, getIdentifyInterceptorCount());
         assertEquals((long)dbHelper.getLastIdentifyInterceptorId(), -1L);
     }
+
+    @Test
+    public void testMultipleIdentifyWithSetActionAndUserIdUpdated() throws JSONException {
+        long [] timestamps = {1000, 1001, 1002, 1003, 1004};
+        clock.setTimestamps(timestamps);
+        ShadowLooper looper = Shadows.shadowOf(amplitude.logThread.getLooper());
+        looper.runToEndOfTasks();
+        amplitude.identify(new Identify().set("key1", "key1-value1").set("key2", "key2-value1").set("key3", "key3-value1"));
+        amplitude.identify(new Identify().set("key1", "key1-value2").set("key4", "key4-value1"));
+        amplitude.identify(new Identify().set("key2", "key2-value2"));
+        amplitude.identify(new Identify().set("key3", "key3-value2").set("key4", "key4-value2"));
+        amplitude.setUserId("identify-user-id");
+        amplitude.identify(new Identify().set("key3", "key3-value3").set("key4", "key4-value3"));
+        looper.runToEndOfTasks();
+        DatabaseHelper dbHelper = DatabaseHelper.getDatabaseHelper(context);
+        assertEquals(1L, getUnsentIdentifyCount());
+        assertEquals(1L, getIdentifyInterceptorCount());
+        assertEquals((long)dbHelper.getLongValue(AmplitudeClient.LAST_IDENTIFY_ID_KEY), 1L);
+        assertEquals((long) dbHelper.getLongValue(AmplitudeClient.SEQUENCE_NUMBER_KEY), 5L);
+        assertEquals((long)dbHelper.getLongValue(AmplitudeClient.LAST_EVENT_TIME_KEY), timestamps[4]);
+
+        looper.runToEndOfTasks();
+        looper.runToEndOfTasks();
+        RecordedRequest request = runRequest(amplitude);
+        JSONArray events = getEventsFromRequest(request);
+        assertEquals(events.length(), 2);
+        JSONObject event = events.getJSONObject(0);
+        assertEquals(event.getString("event_type"), Constants.IDENTIFY_EVENT);
+        assertEquals(event.getLong("event_id"), 1);
+        assertEquals(event.getLong("timestamp"), timestamps[0]);
+        assertEquals(event.getLong("sequence_number"), 1);
+        assertEquals(event.getString("user_id"), "null");
+        JSONObject userProperties = event.getJSONObject("user_properties");
+        assertEquals(userProperties.length(), 1);
+        JSONObject expected = new JSONObject();
+        expected.put("key1", "key1-value2");
+        expected.put("key2", "key2-value2");
+        expected.put("key3", "key3-value2");
+        expected.put("key4", "key4-value2");
+        JSONObject expectedGroups = new JSONObject();
```

Comment:
Is `expectedGroups` being used anywhere in the tests?

Reply:
nvm, that's a copy from other test, cleaning it up